packages/web-runtime/src/components/Topbar/Notifications.vue

```diff
@@ -198,28 +193,34 @@ export default {
           data?.sort((a, b) => (new Date(b.datetime) as any) - (new Date(a.datetime) as any)) || []
       } catch (e) {
         console.error(e)
-      } finally {
-        if (unref(dropdownOpened)) {
-          setAdditionalData()
-        }
-        loading.value = false
       }
     }).restartable()
 
-    const deleteNotifications = async (ids: string[]) => {
-      loading.value = true
+    const deleteNotificationsTask = useTask(function* (signal, ids) {
       try {
-        await clientService.httpAuthenticated.delete(
+        yield clientService.httpAuthenticated.delete(
           'ocs/v2.php/apps/notifications/api/v1/notifications',
           { data: { ids } }
         )
       } catch (e) {
         console.error(e)
       } finally {
         notifications.value = unref(notifications).filter((n) => !ids.includes(n.notification_id))
-        loading.value = false
       }
-    }
+    }).restartable()
+
+    const computeNotificationDataTask = useTask(function* (signal, notifications) {
+      for (const notification of unref(notifications)) {
+        if (!notification.computedMessage) {
+          notification.computedMessage = yield new Promise((resolve) =>
+            resolve(getMessage(notification))
+          )
```

Comment:
This looks superweird - why not just `notification.computedMessage = getMessage(notification)`?

Reply:
sonarcloud is reporting a bug, if you don't use yield in a useTask generator func

Reply:
SonarCloud has a point here, the question is: why use a generator function when you don't have async behaviour? I guess you just want to rely on the `isRunning` functionality of a task? I'd prefer not to use a task here, although I don't have the strongest opinion about it 🤷 A comment would be nice at least, explaining why this is being wrapped in a promise.

Reply:
I really prefer to have a task here, since a loading state seems to be needed here (see previous) implementation. It's quite more convenient to have all loading states originate from the same interface. The previous implementation was hard to understand.

Reply:
moved to fetchNotificationTasj / SSE listener